lib/components/drawer/sliver_search_bar.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+
+class SliverSearchBarWidget extends StatelessWidget {
+  final Function filterBySearchBarText;
```

Comment:
Type this as `Function(String)`.

Also prefix it with `on`, ie `onFilterBySearchBarText`. Event handlers tend to be prefixed with `on` in the standard lib.